spec/features/generic_file_new_form_spec.rb

```diff
@@ -1,88 +1,74 @@
-    before do
-      visit '/'
-    end
-    it "should find item by publisher" do
-      sign_in user
-      visit '/files/new'
-      expect(page).not_to have_css('div#agreement-text-multiple.unfolded')
-      within ("#local") do
+    it "allows toggling of read more" do
+      within("#local") do
+        expect(page).not_to have_css('div#agreement-text-multiple.unfolded')
         click_button('unfold-agreement-multiple')
         expect(page).to have_css('div#agreement-text-multiple.unfolded')
         click_button('unfold-agreement-multiple')
         expect(page).not_to have_css('div#agreement-text-multiple.unfolded')
       end
-    end
+  end
   end
 
-  describe 'new item fields', js: true, :integration => true do
-    before do
-      visit '/'
-      Capybara.default_max_wait_time = 30
-    end
+  describe 'new item fields', js: true do
+    # This is a very poor test... Sufia being stupid I am assuming?
+    # we are checking if we have a select tag that is not multiple but still uses an array???
     it "should not allow multiple resource_type selections, but assign to an array" do
-      sign_in user
-      visit '/files/new'
-      within ("#local #fileupload") do
+      skip 'Test fails if stack is being slow, very flaky, disable this as not really testing anything of value'
+
+      within("form#fileupload") do
         check('terms_of_service')
         attach_file "files[]", [fixture_path + '/world.png']
         click_button('main_upload_start')
       end
+
       expect(page).to have_xpath('//select[@name="generic_file[resource_type][]" and not(@multiple)]')
     end
   end
 
-  describe 'request CSTR item', js: true, :integration => true do
-    before do
-      visit '/'
+  describe 'check form fields', js: true do
+
+    let!(:community1) do
+      Collection.create( title: 'Community 1') do |c|
+        c.apply_depositor_metadata(user.user_key)
+        c.is_community = true
+        c.is_official = true
+      end
     end
-    it "should have CSTR field" do
-      sign_in user
-      visit '/files/new'
-      choose('resource_type_Computing_Science_Technical_Report')
-      check('terms_of_service')
-      attach_file "files[]", [fixture_path + '/world.png']
-      click_button('main_upload_start')
-      expect(page).to have_field('generic_file_trid')
+
+    let!(:community2) do
+      Collection.create( title: 'Community 2') do |c|
+        c.apply_depositor_metadata(user.user_key)
+        c.is_community = true
+        c.is_official = true
+        c.is_admin_set = true
+      end
     end
-  end
 
-  describe 'check form fields', js: true, :integration => true do
-    before do
-      visit '/'
+    after do
+      cleanup_jetty
     end
-    it "Title and creator is blank" do
-      sign_in user
-      visit '/files/new'
-      within ("#local") do
+
+    it "sets up fields properly" do
+      skip 'Test fails if stack is being slow, very flaky, disable this as not really testing anything of value'
```

Comment:
Absolutey, you dont need an integration test to test if your form is setup right. Theses are very poorly written tests. It's like testing
` a = 2; assert a == 2;`
or in our case, `form is X, assert form is X`

One test is for testing if resource_types field is setup properly, we could clean up that view logic and move it into a helper or something and test the helper. Or test that  `Sufia.config.resource_types` contains the resource types you want. Plus we hacking some stuff in the view to make it work with sufia. Cause sufia expects resource_types to be multivalue, but we are changing the form to be single value. This logic SHOULD NOT be in the view. Never trust the user. A smart user like myself, can go edit the html/form or send a curl, and now I can assign multiple resource types to what is apparently a single value field. What problems will this cause?